### ISSUE TITLE
studio: color picker

### ISSUE THREAD

**Issue by peterpeterparker** [deckgo/deckdeckgo#260]
we bet on the platform but for the moment:

1. <input type="color"/> doesn't work on iPad
2. the position of the color popup is a bit random

let's see in the future if the platform/browsers offers more control on the color picker. if not we might have to create our own component

**Comment by bzuillsmith** [comment]
I definitely think a custom component is going to be the way to go in the long run. As an intermediary step, it could be a custom component that simply lets you select from a custom palette. The palette could be based on the colors already used in the presentation.

I imagine it would work by pressing a paint bucket, which may show animated button branch around it with "background" and "text" color. Then clicking one of those would cause several solid colored circles appear around the button. This keeps it all localized to the paint button, keeps with the material design, and really quick for the user to select a color.

A step further would be to have one of the color buttons actually be a "custom" button that opens the color picker if the needed color doesn't already exist in the palette.

Easier said than done, I know! Just throwing out thoughts. Hope they are helpful :)

**Comment by peterpeterparker** [comment]
> The palette could be based on the colors already used in the presentation.

This is an excellent idea! We have somehow the idea to introduce at some point the ability to have themes aka "color palette" for the presentations, that would work well together and would be a reason to go deeper than the platform

Thx for all the valid inputs and good ideas 👍

**Comment by peterpeterparker** [comment]
I've created a rough color picker and als implemented it in our editor (studio) and in the inline WYSIWYG editor. This has been deployed with v1.0.0-beta.5 of the editor

The idea of @bzuillsmith of "animated button branch around it with "background" and "text" color" would have been really fancy but I went for a plain solution where "background" or "text" have to be selected first. Probably a bit lazy but also without such animation the same component was also usable in the "inline editor" which displays a color picker when text is selected

The component itself could be improved (Benjamin go for it) and I also opened a new issue as for the future we should not think "just" in colors but would be cool to be able to defined and use palettes etc. See https://github.com/deckgo/deckdeckgo/issues/350
